portal-kernel/src/com/liferay/portal/kernel/workflow/BaseWorkflowNode.java

```diff
@@ -26,15 +26,21 @@ public String getLabel(Locale locale) {
 			return HtmlUtil.escape(label);
 		}
 
-		label = _labelMap.get(LocaleUtil.getSiteDefault());
+		Language language = LanguageUtil.getLanguage();
+
+		label = language.get(locale, _name);
```

Comment:
@p-albuquerque in the conditional below you should check if the label is different of the name:

```java
if (!Objects.equals(label, _name)) {
	return HtmlUtil.escape(label);
}
```

because when there is no translation for this key (name) the key is the default value, [see](https://github.com/liferay/liferay-portal/blob/902609c54b58c748c24891736d21dd790fee8bb7/portal-impl/src/com/liferay/portal/language/LanguageImpl.java#L926)

or we should pass the default value as null, for example:

```java
label = language.get(locale, _name, null);
```